pkg/skaffold/runner/runcontext/v2/context.go

```diff
@@ -200,25 +207,43 @@ func (rc *RunContext) BuildConcurrency() int                         { return rc
 func (rc *RunContext) IsMultiConfig() bool                           { return rc.Pipelines.IsMultiPipeline() }
 func (rc *RunContext) GetRunID() string                              { return rc.RunID }
 
-// GetRenderOutputPath points to the directory where the manifest rendering happens. By default, it is set to "<WORKDIR>/.kpt-pipeline".
-func (rc *RunContext) GetRenderOutputPath() (string, error) {
+// GetHydrationDir points to the directory where the manifest rendering happens. By default, it is set to "<WORKDIR>/.kpt-pipeline".
+func (rc *RunContext) GetHydrationDir() (string, error) {
 	var hydratedDir string
 	var err error
-	if rc.Opts.RenderOutput == "" {
-		// Workdir is set to pwd and shall always exist.
+
+	if rc.Opts.HydrationDir == constants.DefaultHydrationDir {
 		hydratedDir = filepath.Join(rc.WorkingDir, constants.DefaultHydrationDir)
-	} else if hydratedDir, err = filepath.Abs(rc.Opts.RenderOutput); err != nil {
+	} else {
+		hydratedDir = rc.Opts.HydrationDir
+	}
+	if hydratedDir, err = filepath.Abs(hydratedDir); err != nil {
 		return "", err
 	}
+
 	if _, err := os.Stat(hydratedDir); os.IsNotExist(err) {
-		logrus.Infof("render output path does not exist, creating %v\n", hydratedDir)
+		logrus.Infof("hydrated-dir does not exist, creating %v\n", hydratedDir)
 		if err := os.MkdirAll(hydratedDir, os.ModePerm); err != nil {
 			return "", err
 		}
+	} else if !isDirEmpty(hydratedDir) {
+		if !rc.Opts.AssumeYes {
+			if ok := ackUser(os.Stdin); !ok {
+				cmd.Exit(nil)
+			}
+		}
```

Comment:
It feels odd to have this prompting in the runcontext.  It means that we can't call this function without fear that it may result in a user prompt.

Reply:
changed to a standalone func and made user prompt a passed-in arg.

Reply:
Modified the user behavior a little bit.
1. if user doesn't specify the `--hydration-dir`, it means they are not manipulating the dry/wet manifests. In such case, skaffold does not prompt any message
2. If user specified the `--hydration-dir` and if the dir is not empty, skaffold prompts the message, asking them to confirm. Also, skaffold lets them know that they can use `--assume-yes=true` to skip this prompt.
3. For developers who use this function, since it's not tied to a runcontext, developers can specify the skaffoldOptions as well as a working directory (in previous version, I assume the workingDir can't be empty and has been initialized in GetRuntimeContext). Also, they shall be aware of the prompt message from the promptIfNeededd arg.

WDYT?    @briandealwis